### PR TITLE
Delete endpoint

### PR DESCRIPTION
**Pull request by elysableu**
- [x] Wrote tests?
- [x] Implemented the code?
- [x] Reviewed my code?

Necessary check marks:

  - [x] All Tests are Passing
  - [ ] The code will run locally

Type of change

  - [x] New feature
  - [ ] Bug Fix

Implements/Fixes:

  # Description of work...
- Added the delete endpoint for merchants and items
- Added tests for delete endpoint for merchants and items
- Removed unnecessary methods from serializers

Check the correct boxes

  - [x] This broke nothing
  - [ ] This broke some stuff
  - [ ] This broke everything

Testing Changes

  - [ ] No Tests have been changed
  - [x] Some Tests have been changed
  - [ ] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

  - [x] My code has no unused/commented out code
  - [x] I have reviewed my code
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have fully tested my code

(Optional) What questions do you have? Anything specific you want feedback on?

  * 

(For Fun!) Please include a link to a gif [or add an emoji] of your feelings about this branch.

Link:

### CODE REVIEW

File: app/serializers/customer_serializer.rb [hunk 1]
```diff
@@ -1,8 +1,4 @@
 class CustomerSerializer
   include JSONAPI::Serializer
   attributes :first_name, :last_name
-
-  def id
-    object.id.to_s
-  end
```
Comment (DustinPeukert) [inline app/serializers/customer_serializer.rb:7]: Thank you for removing these!

File: spec/requests/api/v1/items_request_spec.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@
   end
   
   describe "POST /item" do
-    it 'can create a poster' do
```
Comment (DustinPeukert) [inline spec/requests/api/v1/items_request_spec.rb:39]: Nice catch on this mistake!

File: spec/requests/api/v1/items_request_spec.rb [hunk 3]
```diff
@@ -60,4 +60,19 @@
...
+      merchant = Merchant.create( name: "Lula Faye")
+      item1 = Item.create(  name: "Crochet Hook", description: "5mm hook", unit_price: 5.99, merchant_id: merchant.id )
+      item2 = Item.create( name: "Cashmere Yarn", description: "A teal green yarn", unit_price: 19.99, merchant_id: merchant.id )
+
+      expect(Item.all.length).to eq(2)
+
+      delete "/api/v1/items/#{item1.id}"
+
+      expect(Item.all.length).to eq(1)
+      expect{ Item.find(item1.id) }.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (DustinPeukert) [inline spec/requests/api/v1/items_request_spec.rb:75]: Love these well rounded tests! Smart thinking to check for a raised error when trying to find the deleted item!

**Review by DustinPeukert — APPROVED** [review]
These look great Elysa! Especially the tests: they're rounded, readable, and make sense!

File: spec/requests/api/v1/items_request_spec.rb (hunk 3 above)
Comment (elysableu) [reply spec/requests/api/v1/items_request_spec.rb:75]: Why thank you!  I found it in the lesson "Building an API" and it seemed like a useful test to include😅

**Comment by elysableu** [comment]
I'll work to add edge cases as my next task

**Review by JonoSommers — APPROVED** [review]
Tests look great, and everything in the controllers looks correct to me. I will go ahead and merge.
